scipy/optimize/tests/test_optimize.py

```diff
@@ -937,6 +936,10 @@ def test_respect_maxiter(self, method):
         if hasattr(sol, 'njev'):
             assert sol.njev >= sf.ngev
 
+        # method specific tests
```

Comment:
I'm wondering if we should set line 934 to `sol.nit == MAXITER`. I suspect tnc would then fail tests.

Reply:
Oh. you are right.  The test failed. sol.nit is 1, not 4.

> ============================================================================================= FAILURES ==============================================================================================
> ___________________________________________________________________________ TestOptimizeSimple.test_respect_maxiter[TNC] ____________________________________________________________________________
> scipy/optimize/tests/test_optimize.py:935: in test_respect_maxiter
>     assert sol.nit == MAXITER
> E   assert 1 == 4
> E    +  where 1 =      fun: 8.799939396486105\n     jac: array([-1.9414383 , -1.34732119,  0.37293528,  0.38459742, -0.09909762,\n        ...08, 0.0121846 , 0.0121846 , 0.0121846 , 0.00979179,\n       0.0121846 , 0.0121846 , 0.0121846 , 0.0145774 , 0.02632083]).nit

Reply:
I changed "maxiter" to "maxfun" in options of tnc solver.